framework/customizations/reports.py

```diff
@@ -0,0 +1,191 @@
+    sender = "ocs-ci@redhat.com"
+    msg = MIMEMultipart("alternative")
+    rpm_name = get_noobaa_sa_rpm_name()
+    msg["Subject"] = f"noobaa standalone results {rpm_name}"
+    msg["From"] = sender
+    msg["To"] = ", ".join(recipients)
+
+    soup = create_results_html(session)
+    part1 = MIMEText(soup, "html")
+    msg.attach(part1)
+    # with open("/home/oviner/ClusterPath/test7.html", "w") as file:
+    #     # Write the data to the file
+    #     file.write(msg.as_string())
+    try:
+        s = smtplib.SMTP(config.REPORTING["email"]["smtp_server"])
+        s.sendmail(sender, recipients, msg.as_string())
+        s.quit()
+        log.info(f"Results have been emailed to {recipients}")
+    except Exception as e:
+        log.exception(f"Sending email with results failed! {e}")
+
+
+def create_results_html(session):
+    """
+    Add squad analysis to the html test results used in email reporting
+
+    Args:
+        session (obj): Pytest session object
+
+    """
+    failed_tests = []
+    passed_tests = []
+    skipped_tests = []
+    # sort out passed, failed and skipped test cases
+    for result in session.results.values():
+        elapsed_time = f"{int(result.stop - result.start)} sec"
+        if result.passed:
+            passed_tests.append((result.nodeid, elapsed_time, result.longreprtext))
+        elif result.failed:
+            failed_tests.append((result.nodeid, elapsed_time, result.longreprtext))
+        elif result.skipped:
+            skipped_tests.append((result.nodeid, elapsed_time, result.longreprtext))
+        current_dir = Path(__file__).parent.parent.parent
+        html_template = os.path.join(
+            current_dir, "templates", "html_reports", "html_template.html"
+        )
+    total_tests = len(failed_tests) + len(passed_tests) + len(skipped_tests)
+    if total_tests == 0:
+        return
+    passed_percentage = f"{float((len(passed_tests) / total_tests) * 100):.2f}%"
+    failed_percentage = f"{float((len(failed_tests) / total_tests) * 100):.2f}%"
+    skipped_percentage = f"{float((len(skipped_tests) / total_tests) * 100):.2f}%"
+
+    with open(html_template) as fd:
+        html_data = fd.read()
+    soup = BeautifulSoup(html_data, "html.parser")
+
+    # Insert versions into the versions table
+    def add_version_data_to_table(versions, table_id):
+        tbody = soup.find(id=table_id)
+        for component, version in versions.items():
+            row = soup.new_tag("tr")
+
+            # Component name cell
+            component_cell = soup.new_tag("td")
+            component_cell.string = component
+            row.append(component_cell)
+
+            # Version cell
+            version_cell = soup.new_tag("td")
+            version_cell.string = version
+            row.append(version_cell)
+
+            # Add the row to the table
+            tbody.append(row)
+
+    from utility.utils import get_noobaa_sa_rpm_name
```

Comment:
move to import section

Reply:
it failed in import section

Reply:
please move to start of this function and add comment why import exists in this function instead of starting of file